### ISSUE TITLE
Android manifest merger exceptions are dropped due to an unrelated crash from a Guava incompatibility

### ISSUE THREAD

**Issue by jin** [bazelbuild/bazel#6671]
### Description of the problem / feature request:

Bazel does not propagate messages from exceptions in `ManifestMergerAction` properly.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

See https://github.com/jin/repro/tree/master/6645

The Bazel error is:

```

ERROR: /Users/jingwen/code/repro/6645/src/main/BUILD:3:1: Merging manifest for //src/main:app failed (Exit 1) ResourceProcessorBusyBox failed: error executing command bazel-out/host/bin/external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/ResourceProcessorBusyBox --tool MERGE_MANIFEST -- --manifest src/main/AndroidManifest.xml ... (remaining 8 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
Exception in thread "main" java.lang.NoSuchMethodError: com.google.common.base.Objects.firstNonNull(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
	at com.android.ide.common.res2.MergingException$Builder.build(MergingException.java:129)
	at com.android.manifmerger.XmlElement.<init>(XmlElement.java:140)
	at com.android.manifmerger.XmlElement.initMergeableChildren(XmlElement.java:871)
	at com.android.manifmerger.XmlElement.<init>(XmlElement.java:171)
	at com.android.manifmerger.XmlElement.initMergeableChildren(XmlElement.java:871)
	at com.android.manifmerger.XmlElement.<init>(XmlElement.java:171)
	at com.android.manifmerger.XmlDocument.getRootNode(XmlDocument.java:231)
	at com.android.manifmerger.ManifestMerger2.loadLibraries(ManifestMerger2.java:589)
	at com.android.manifmerger.ManifestMerger2.merge(ManifestMerger2.java:126)
	at com.android.manifmerger.ManifestMerger2.access$1100(ManifestMerger2.java:58)
	at com.android.manifmerger.ManifestMerger2$Invoker.merge(ManifestMerger2.java:1163)
	at com.google.devtools.build.android.AndroidManifestProcessor.mergeManifest(AndroidManifestProcessor.java:181)
	at com.google.devtools.build.android.ManifestMergerAction.main(ManifestMergerAction.java:215)
	at com.google.devtools.build.android.ResourceProcessorBusyBox$Tool$11.call(ResourceProcessorBusyBox.java:126)
	at com.google.devtools.build.android.ResourceProcessorBusyBox.processRequest(ResourceProcessorBusyBox.java:234)
	at com.google.devtools.build.android.ResourceProcessorBusyBox.main(ResourceProcessorBusyBox.java:197)
Target //src/main:app failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 85.869s, Critical Path: 23.21s
INFO: 239 processes: 206 darwin-sandbox, 33 worker.
FAILED: Build did NOT complete successfully
```

Note:

```
Exception in thread "main" java.lang.NoSuchMethodError: com.google.common.base.Objects.firstNonNull(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
```

This is because the `MergingException` class file was compiled against Guava 17, where `firstNonNull` still existed. Since them, it has been moved to the `MoreObjects` class, which doesn't exist from Guava 18 onwards.

The problem here is that Bazel's vendored dependency has moved on to v25. Internally, we patch `MergingException` to use `MoreObjects.firstNonNull`, but we'll need to do something else here as we vendor the sdk_common jars directly.

Also see https://github.com/bazelbuild/bazel/issues/6645 for more context.

**Comment by jin** [comment]
May be related:

https://github.com/bazelbuild/bazel/issues/3190

**Comment by jin** [comment]
Down prioritizing this as #6645 will be fixed by #6970, but as time goes on we'll need to update the android_common jars. Keeping this open to keep track of this issue.

**Comment by nkoroste** [comment]
Any estimates on which release this will go in?
